src/server/controllers/changePassword.ts

```diff
@@ -20,6 +20,10 @@ import {
 } from '@/server/lib/encryptedStateCookie';
 import { ApiError } from '@/server/models/Error';
 
+type ValidPassworUrls = '/reset-password' | '/set-password' | '/welcome';
+
+type ValidResendEmailUrls = '/reset' | '/welcome' | '/set-password';
```

Comment:
I suggest inheriting from `RouteProps` here, using `Extract` for even more type safety as it has to get the strings from RoutePaths, so if they were to change this type would show an error: https://www.typescriptlang.org/docs/handbook/utility-types.html#extracttype-union

Something like this could work:
```suggestion
type ValidPassworUrls = Extract<RoutePaths, '/reset-password', '/set-password', '/welcome'>;

type ValidResendEmailUrls = Extract<RoutePaths, '/reset', '/welcome', '/set-password'>;
```

Reply:
Nice !

Reply:
The syntax was 

```ts 
type ValidResendEmailUrls = Extract< '/reset' | '/welcome' | '/set-password',  RoutePaths>;
``` 

but it works a charm